org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/resource/Storage2UriMapperImpl.java

```diff
@@ -215,10 +215,14 @@ protected boolean isHandled(IFolder folder) {
 	}
 
 	private Iterable<Pair<IStorage, IProject>> getStorages(/* @NonNull */ URI uri, IFile file) {
+		Iterable<Pair<IStorage, IProject>> storages = null;
 		if (file == null || !file.isAccessible()) {
-			return contribution.getStorages(uri);
+			storages = contribution.getStorages(uri);
+		}
+		if (file!=null && (storages == null || Iterables.isEmpty(storages))) {
+			storages = Collections.singleton(Tuples.<IStorage,IProject>create(file, file.getProject()));
```

Comment:
Changing the contract of this method will affect a lot of other locations. Why are you doing this here?